src/main/java/uk/ac/ic/wlgitbridge/bridge/Bridge.java

```diff
@@ -110,62 +114,109 @@ public void startSwapJob() {
         swapJob.start();
     }
 
-    /* TODO: Remove these when WLBridged is moved into RepoStore */
-    public void lockForProject(String projectName) {
-        lock.lockForProject(projectName);
-    }
-
-    public void unlockForProject(String projectName) {
-        lock.unlockForProject(projectName);
-    }
-
-    public boolean repositoryExists(Credential oauth2, String projectName)
-            throws ServiceMayNotContinueException, GitUserException {
-        lockForProject(projectName);
-        GetDocRequest getDocRequest = new GetDocRequest(oauth2, projectName);
-        getDocRequest.request();
-        try {
+    public boolean projectExists(
+            Credential oauth2,
+            String projectName
+    ) throws ServiceMayNotContinueException,
+             GitUserException {
+        try (LockGuard __ = lock.lockGuard(projectName)) {
+            GetDocRequest getDocRequest = new GetDocRequest(
+                    oauth2,
+                    projectName
+            );
+            getDocRequest.request();
             getDocRequest.getResult().getVersionID();
+            return true;
         } catch (InvalidProjectException e) {
             return false;
-        } finally {
-            unlockForProject(projectName);
         }
-        return true;
     }
 
-    public void getWritableRepositories(
+    public void updateRepository(
             Credential oauth2,
             ProjectRepo repo
-    ) throws IOException,
-             GitUserException {
-        Log.info("[{}] Fetching", repo.getProjectName());
-        updateProjectWithName(oauth2, repo);
+    ) throws IOException, GitUserException {
+        String projectName = repo.getProjectName();
+        try (LockGuard __ = lock.lockGuard(projectName)) {
+            Log.info("[{}] Updating", projectName);
+            updateRepositoryCritical(oauth2, repo);
+        }
     }
 
-    public void
-    putDirectoryContentsToProjectWithName(Credential oauth2,
-                                          String projectName,
-                                          RawDirectory directoryContents,
-                                          RawDirectory oldDirectoryContents,
-                                          String hostname)
-            throws SnapshotPostException, IOException, ForbiddenException {
-        lock.lockForProject(projectName);
-        CandidateSnapshot candidate = null;
-        try {
-            Log.info("[{}] Pushing", projectName);
-            String postbackKey = postbackManager.makeKeyForProject(projectName);
-            Log.info(
-                    "[{}] Created postback key: {}",
+    private void updateRepositoryCritical(
```

Comment:
It's supposed to be the "critical section" of methods that acquire locks. Do you think it needs changing?